x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/plan/logical/command/sys/SysTables.java

```diff
@@ -77,8 +77,11 @@ public final void execute(SqlSession session, ActionListener<SchemaRowSet> liste
         // namely one param specified with '%', everything else empty string
         // https://docs.microsoft.com/en-us/sql/odbc/reference/syntax/sqltables-function?view=ssdt-18vs2017#comments
 
-        if (clusterPattern != null && clusterPattern.pattern().equals(SQL_WILDCARD)) {
-            if ((pattern == null || pattern.pattern().isEmpty()) && CollectionUtils.isEmpty(types)) {
+        // catalog enumeration
+        if (clusterPattern == null || (clusterPattern != null && clusterPattern.pattern().equals(SQL_WILDCARD))) {
+            // enumerate only if pattern is "" and no types are null
+            if (pattern != null && StringUtils.EMPTY.equals(pattern.pattern()) && index == null
```

Comment:
Made me think of `StringUtils.isEmpty/hasText` which matches `"  "` no empty strings.

Reply:
I've removed it since I was also trying to use it directly to do string comparison (and thus avoid the null check) however since the string is embedded the parent needs to exist - long story short, it's not needed.